src/main/java/com/minecolonies/coremod/entity/ai/mobs/AbstractArcherBarbarian.java

```diff
@@ -0,0 +1,302 @@
+    {
+        super.onDeath(cause);
+
+        if (cause.getSourceOfDamage() instanceof EntityArrow && cause.getEntity() instanceof EntityPlayer)
+        {
+            final EntityPlayer entityplayer = (EntityPlayer) cause.getEntity();
+            final double d0 = entityplayer.posX - this.posX;
+            final double d1 = entityplayer.posZ - this.posZ;
+
+            final double distance = 2500.0D;
+
+            if (d0 * d0 + d1 * d1 >= distance)
+            {
+                entityplayer.addStat(AchievementList.SNIPE_SKELETON);
+            }
+        }
+    }
+
+    /**
+     * Gives armor or weapon for entity based on given DifficultyInstance
+     */
+    protected void setEquipmentBasedOnDifficulty(final DifficultyInstance difficulty)
+    {
+        super.setEquipmentBasedOnDifficulty(difficulty);
+        this.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.BOW));
+    }
+
+    /**
+     * Called only once on an entity when first time spawned, via egg, mob spawner, natural spawning etc, but not called
+     * when entity is reloaded from nbt. Mainly used for initializing attributes and inventory
+     */
+    @Nullable
+    public IEntityLivingData onInitialSpawn(final DifficultyInstance difficulty, @Nullable final IEntityLivingData living)
+    {
+        final IEntityLivingData livingdata = super.onInitialSpawn(difficulty, living);
+        this.setEquipmentBasedOnDifficulty(difficulty);
+        this.setEnchantmentBasedOnDifficulty(difficulty);
+        this.setCombatTask();
+        this.setCanPickUpLoot(this.rand.nextFloat() < 0.55F * difficulty.getClampedAdditionalDifficulty());
+
+        if (this.getItemStackFromSlot(EntityEquipmentSlot.HEAD).isEmpty())
+        {
+            final Calendar calendar = this.world.getCurrentDate();
+
+            if (calendar.get(2) + 1 == 10 && calendar.get(5) == 31 && this.rand.nextFloat() < 0.25F)
+            {
+                this.setItemStackToSlot(EntityEquipmentSlot.HEAD, new ItemStack(this.rand.nextFloat() < 0.1F ? Blocks.LIT_PUMPKIN : Blocks.PUMPKIN));
+                this.inventoryArmorDropChances[EntityEquipmentSlot.HEAD.getIndex()] = 0.0F;
+            }
+        }
+
+        return livingdata;
+    }
+
+    /**
+     * sets this entity's combat AI.
+     */
+    public void setCombatTask()
+    {
+        if (this.world != null && !this.world.isRemote)
+        {
+            this.tasks.removeTask(this.aiAttackOnCollide);
+            this.tasks.removeTask(this.aiArrowAttack);
+            final ItemStack itemstack = this.getHeldItemMainhand();
+
+            if (itemstack.getItem() == Items.BOW)
+            {
+                int i = BOW_DAMAGE;
+
+                if (this.world.getDifficulty() != EnumDifficulty.HARD)
+                {
+                    i = BOW_DAMAGE_HARD;
+                }
+
+                this.aiArrowAttack.setAttackCooldown(i);
+                this.tasks.addTask(4, this.aiArrowAttack);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)